internal/service/vpc/datasource_project_vpc.go

```diff
@@ -28,34 +65,30 @@ func DatasourceProjectVPC() *schema.Resource {
 }
 
 func datasourceProjectVPCRead(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var splitId []string
+	var vpcId string
 	client := m.(*aiven.Client)
-
 	projectName := d.Get("project").(string)
 	cloudName := d.Get("cloud_name").(string)
-	vpcId, hasId := d.GetOk("id")
+	if id, hasId := d.GetOk("vpc_id"); hasId {
+		var err error
+		splitId, err = schemautil.SplitResourceID(id.(string), 2)
+		if err != nil {
+			return diag.Errorf("error splitting vpc_id: %s:", err)
+		}
+
+		projectName = splitId[0]
+		vpcId = splitId[1]
+	}
 
 	vpcs, err := client.VPCs.List(projectName)
 	if err != nil {
-		return diag.FromErr(err)
+		return diag.Errorf("error getting a list of project VPCs: %s", err)
 	}
 
 	filteredVPCs := make([]*aiven.VPC, 0)
-
 	for _, vpc := range vpcs {
-		condition := vpc.CloudName == cloudName
-
-		// We don't need to care about the cloud name if the ID is provided
-		if hasId {
-			splitId, err := schemautil.SplitResourceID(vpcId.(string), 2)
-			if err != nil {
-				return diag.FromErr(err)
-			}
-
-			id := splitId[1]
-			condition = vpc.ProjectVPCID == id
-		}
-
-		if condition {
+		if vpc.CloudName == cloudName || (vpcId != "" && vpc.ProjectVPCID == vpcId) {
 			filteredVPCs = append(filteredVPCs, vpc)
```

Comment:
What if cloudName and vpc.CloudName are both empty? Is that possible? Or should it be validated separately:
```go
if cloudName == "" && vpcId == "" {
    // one must be provided
}

if vpc.CloudName == cloudName || vpc.ProjectVPCID == vpcId {
    filteredVPCs = append(filteredVPCs, vpc)
}
```